### PR TITLE
add simulator test

### PR DESCRIPTION
**Pull request by silkeholmebonnen**
This PR should close issue #38. 

The test makes sure that there are fewer than 10 errors in total, and checks that no errors of type `tweet`, `register`, `follow` or `unfollow`. This is to get around the fact that we usually get a few `ReadTimeout` and `ConnectionError`.  I also limited the simulator to only do a 1000 requests. I don't know if we want this or if we want to fire the 25000 requests of?

### CODE REVIEW

**Review by A-Guldborg — CHANGES_REQUESTED** [review]
I think we need to reduce the amount of docker images and try to use the main docker image we have which will mimick production more. Instead have steps in our workflow files that uses the main docker image and then pointing our tests towards that one / running our tests in that docker image.

Also, are we certain we cannot fix the errors (`ReadTimeout` and `ConnectionError`) that we get from the simulator? I mean, it seems we might handle many simultaneous requests badly so perhaps we should look into that? (In a separate PR possibly)

I don't know what you are thinking about these two points and if there could be any issues with that? Anyways, a couple things that I at least think we need to fix with this PR regardless of which way we decide to test and how we consider the two errors types :)

File: simapi/minitwit_simulator_test.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    # Check each line for the presence of "tweet"
+    for line in lines:
+        assert "register" not in line, "Found a line containing 'register'"
+
+
+def test_no_follow_errors(log_file):
+    # Read all lines from the file
+    lines = log_file.readlines()
+
+    # Check each line for the presence of "tweet"
```
[the remark this answers is not in the document]
Comment (silkeholmebonnen) [reply simapi/minitwit_simulator_test.py]: I removed all comments. I don't think any comments are necessary. 

I guess we could do it the other way around as well, but this way makes more sense to me at least.

File: simapi/Dockerfile-api-test [hunk 2]
```diff
@@ -2,13 +2,11 @@ FROM python:3.7-alpine
...
 COPY requirements.txt /minitwit-api-tests/
 
 WORKDIR /minitwit-api-tests
 
 RUN pip install -r requirements.txt
 
-EXPOSE 5001
-
-ENTRYPOINT ["pytest", "-s", "minitwit_sim_api_test.py"]
+ENTRYPOINT pytest -s minitwit_sim_api_test.py && python ./minitwit_simulator.py "http://minitwit-api:5001" "./minitwit_scenario.csv" true > ./simulator_output.txt && pytest ./minitwit_simulator_test.py "./simulator_output.txt"
```
[the remark this answers is not in the document]
Comment (silkeholmebonnen) [reply simapi/Dockerfile-api-test:12]: 1. I think the other one is better practice but I didn't know how else to do it
2. First the API tests are run. Afterwards the simulator runs and the outcome of the simulator is tested. This is probably not the best way to do it, but it works. But any other ways of doing it are very welcome:)
3. As far as I understand we did not need to expose 5001. This is just the test image so why would we expose it?

[the remark this answers is not in the document]
Comment (A-Guldborg) [reply simapi/Dockerfile-api-test:12]: For 1. and 2.: If it works I agree lets use it :)
For 3.: Sure, I don't know why we exposed it in the first place but you are probably right! :)

**Pull request merged**